### DISCUSSION TITLE
What can I do to troubleshoot high Steno packet loss?

### DISCUSSION THREAD

**Discussion by w4rc0n** [Security-Onion-Solutions/securityonion#5322] [Unsupported Versions] [answered]
I have a distributed set up including:
- 1 manager node
- 1 sensor node
- 1 search node

On the sensor, I have 0% packet loss with both Zeek and Suricata, but have 89% packet loss for Stenographer. They are writing to HDD's in RAID10, which I assume could be a bottleneck but not to this extent. The only thing I could find for troubleshooting packet loss in the docs was in regards to Zeek and Suricata and setting more workers, however Steno doesn't function in the same way. Here is what Grafana looks like for Disk on that sensor:

![image](https://user-images.githubusercontent.com/49037260/131157589-31562e3b-0d37-47b3-a8ac-0e889da48959.png)

Here is the monitoring interface stats for reference on the amount of packets coming in. 

![image](https://user-images.githubusercontent.com/49037260/131157808-5904ec84-66f8-4e43-a12c-4d8b40142e77.png)

Can anyone point me in the right direction here?

**Comment by TOoSmOotH** [comment, marked as answer]
Looks to be an IO limitation that is causing the drops

**Reply by w4rc0n** [reply]
@TOoSmOotH I thought so too, and I went deep down that rabbit hole but it really doesn't seem to be disk I/O
![image](https://user-images.githubusercontent.com/49037260/142486649-305ff2f7-d09d-4662-9c0e-8cb56684e554.png)

70-80% idle with no io wait. 20-25M/s is far under the max throughput of WD Blacks in RAID10.

**Reply by TOoSmOotH** [reply]
Are you using XFS?

**Reply by w4rc0n** [reply]
Yes

**Reply by TOoSmOotH** [reply]
2.3.90 will help then. It will also improve the tracking fo the actual packet loss.
